src/iqm/benchmarks/optimization/qscore.py

```diff
@@ -622,10 +696,107 @@ def __init__(self, backend_arg: IQMBackendBase, configuration: "QScoreConfigurat
                 list(x) for x in cast(Sequence[Sequence[int]], configuration.custom_qubits_array)
             ]
 
+    def greedy_vertex_cover_with_mapping(self, G: nx.Graph):
+        """
+        Approximate minimum vertex cover with mapping of nodes -> edges they cover.
+
+        Args:
+            G (nx.Graph): input graph
+
+        Returns:
+            cover_map (dict): {node: [(u,v), ...]} showing which edges each node covers
+        """
+
+        G = G.copy()
+        cover_map = {}
+
+        while G.number_of_edges() > 0:
+            # Pick node with max degree
+            node = max(G.degree, key=lambda x: x[1])[0]
+
+            # Collect neighbors (unique)
+            neighbors = list(G.neighbors(node))
+            cover_map[node] = neighbors
+
+            # Remove node (and incident edges)
+            G.remove_node(node)
+
+        return cover_map
+
+    def generate_maxcut_ansatz_star(  # pylint: disable=too-many-branches
+        self,
+        graph,
+        theta,
+    ):
+        """Generate an ansatz circuit for QAOA MaxCut, with measurements at the end.
+
+        Args:
+            graph (networkx graph): the MaxCut problem graph
+            theta (list[float]): the variational parameters for QAOA, first gammas then betas
+
+        Returns:
+            QuantumCircuit: the QAOA ansatz quantum circuit.
+        """
+
+        gamma = theta[: self.num_qaoa_layers]
+        beta = theta[self.num_qaoa_layers :]
+
+        if self.graph_physical.number_of_nodes() != graph.number_of_nodes():
+            num_qubits = self.graph_physical.number_of_nodes()
+            # re-label the nodes to be between 0 and _num_qubits
+            self.node_to_qubit = {node: qubit for qubit, node in enumerate(list(self.graph_physical.nodes))}
+            self.qubit_to_node = dict(enumerate(list(self.graph_physical.nodes)))
+        else:
+            num_qubits = graph.number_of_nodes()
+            self.node_to_qubit = {node: node for node in list(self.graph_physical.nodes)}  # no relabeling
+            self.qubit_to_node = self.node_to_qubit
+
+        covermap = self.greedy_vertex_cover_with_mapping(self.graph_physical)
+        new_covermap = {}
+        for key, value in covermap.items():
+            new_covermap[self.node_to_qubit[key]] = [self.node_to_qubit[i] for i in value]
+        covermap = new_covermap
+
+        compr = QuantumRegister(1, "compr")
+        q = QuantumRegister(num_qubits, "q")
+        c = ClassicalRegister(num_qubits, "c")
+        qaoa_qc = IQMCircuit(compr, q, c)  # num_qb+1,num_qb)
+        # in case the graph is trivial: return empty circuit
+        if num_qubits == 0:
+            return QuantumCircuit(1)
+        for i in range(1, num_qubits):
+            qaoa_qc.h(i)
```

Comment:
```suggestion
        for i in range(1, num_qubits+1):
            qaoa_qc.h(i)
```
I think the loop should probably go to num_qubits+1 to also apply a Hadamard on the last qubit. In an example circuit I checked it looked like a Hadamard was missing: 
<img width="850" height="1020" alt="image" src="https://github.com/user-attachments/assets/a7b7f34b-d673-472b-855a-1d03dda68c24" />

Reply:
AH good catch! Fixed it now